src/extension_utils.ts

```diff
@@ -12,14 +12,52 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 import * as yaml from 'yaml';
-import { readFileSync } from 'fs';
+import { satisfies, minVersion, parse } from 'semver';
+import { isString } from 'util';
+
+const dependencyKey = 'dependencies';
+const devDependencyKey = 'dev_dependencies';
+
+function getDependencyVersion(document:yaml.Document, dependency:string, isADevDependency:boolean = false):string|undefined {
+    const key = isADevDependency ? devDependencyKey : dependencyKey;
+    if (document.contents.has(key)) {
+        const dependencyCollection = document.contents.get(key);
+        if (dependencyCollection !== null && dependencyCollection.has(dependency)) {
+            const collectedDependency = dependencyCollection.get(dependency);
+    
+            // Simple key value dependencies (e.g. not hosted ones) are simple
+            // strings here. Otherwise, we need to do `get` one more time.
+            if (isString(collectedDependency)) {
+                return collectedDependency;
+            } else {
+                return collectedDependency.get('version');
+            }
+        }
+    }
+
+    return undefined;
+}
+
+// Verifies that a provided dependency is present and has a minimum version that matches a given dependency range.
+//
+// Returns false if the dependency does not exist, is invalid, or is below the provided range.
+// Expects `dependency` to be the exact dependency string, `qualifyingRange` to be a compatible semver range.
```

Comment:
Note that the range part is important. I originally tried to make it just the minimum version, but the semver package doesn't have a good way to compute "greater than or equal to". It only has "greater than". Comparatively, a range is super straight forward.